File: services/reservation/src/main/java/com/example/iter/reservation/domain/entity/Receipt.java
package com.example.iter.reservation.domain.entity;

import com.example.iter.common.entity.BaseCreatedAtEntity;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

// ERD RECEIPT 엔티티 — 수령 시점 상태 기록 (Rental과 같은 도메인이므로 직접 연관관계 사용)
@Entity
@Table(name = "receipt")
@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class Receipt extends BaseCreatedAtEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "rental_id", nullable = false, unique = true)
    private Rental rental;

    @Enumerated(EnumType.STRING)
    @Column(name = "product_condition", nullable = false, length = 20)
    private ProductConditionType productCondition;

    @Column(name = "condition_detail", columnDefinition = "TEXT")
    private String conditionDetail;

    @Column(name = "received_at")
    private LocalDateTime receivedAt;
}

File: services/reservation/src/main/kotlin/com/example/iter/reservation/domain/entity/Receipt.kt
package com.example.iter.reservation.domain.entity

import com.example.iter.common.entity.BaseCreatedAtEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.OneToOne
import jakarta.persistence.Table

import java.time.LocalDateTime

// ERD RECEIPT 엔티티 — 수령 시점 상태 기록 (Rental과 같은 도메인이므로 직접 연관관계 사용)
@Entity
@Table(name = "receipt")
class Receipt @JvmOverloads constructor(

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "rental_id", nullable = false, unique = true)
    val rental: Rental,

    @Enumerated(EnumType.STRING)
    @Column(name = "product_condition", nullable = false, length = 20)
    val productCondition: ProductConditionType,

    @Column(name = "condition_detail", columnDefinition = "TEXT")
    val conditionDetail: String? = null,

    @Column(name = "received_at")
    val receivedAt: LocalDateTime? = null,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

) : BaseCreatedAtEntity()
